lib/logging/lib/logging.c

```diff
@@ -29,8 +29,7 @@ static merr_stringify *log_err_ctx_stringify;
 static thread_local char log_buffer_tls[1024];
 static uint64_t log_squelch_ns = LOG_SQUELCH_NS_DEFAULT;
 
-static void HSE_PRINTF(2, 3)
-backstop(const int level, const char *const fmt, ...)
+static void HSE_PRINTF(2, 3) backstop(const int level, const char *const fmt, ...)
```

Comment:
Seems to be something up here.

Reply:
It doesn't do well with our attribute macros that take arguments.

Reply:
There are a few options to try it seems.

- `AttributeMacros`
- `StatementAttributeLikeMacros`
- `StatementMacros`
- `TypenameMacros`

Using some combo of these I bet we can get the desired behavior

Reply:
It works better when `HSE_PRINTF()` is before the function type:
```
HSE_PRINTF(2, 3)
static void
backstop(const int level, const char *const fmt, ...)
{
    ...
```

Reply:
I noticed you can also put it before the void, but I would be interested to see what it could output with the above options set.

Reply:
TypenameMacros seems to solve this issue.